src/LoginDialog/OidcLoginButtons.tsx

```diff
@@ -22,7 +22,7 @@ export const OidcLoginButtons: React.FC = () => {
         const { links } = await getOidcProfile(authenticationRealmId, profile.id)
         const baseRedirectUrl = links['authorization-endpoint']
 
-        window.location.href = generateOidcLoginRedirectUrl(baseRedirectUrl, cId, location.pathname);
+        window.location.href = await generateOidcLoginRedirectUrl(baseRedirectUrl, cId, location.pathname);
```

Comment:
Was there a delay while waiting for this promise to resolve?

Reply:
I don't fully understand your question.

The reason this is changed is to match the types, and with the previous `await` call on line 22, I didn't worry about slowness, since I believe the previous call is _actually_ a blocking network call.

This change is sort of the knock on effect, of calling, `window.crypto.subtle.digest()` which returns a promise. That call _can_ take a long time which is why it returns a promise. It _should_ be fairly quick for us, based on the short amount of input data we are using. 

This benchmark: https://www.measurethat.net/Benchmarks/Show/1246/0/sha256 suggests that we can get 100,000 operations per second, over the digest call for 1K of data (we are only doing 100 bytes), on my laptop, and we onyl need to call it once.

Reply:
If this is a dependent URL required for the app to function, I'm okay with us awaiting here. It's more that if generateOidcLoginRedirectUrl() took a noticeably long time to run, then we'd have to ensure it doesn't lock up the UI or negatively impact the UX.